Monika After Story/game/zz_weather.rpy

```diff
@@ -99,16 +99,59 @@ image mas_island_night = "mod_assets/location/special/night_without_frame.png"
 default persistent._mas_weather_MWdata = {}
 # stores locked/unlocked status for weather
 
+default persistent._mas_force_weather = False
+
+#When did we last check if it could rain
+default persistent._mas_date_last_checked_rain = None
+
+#Should it rain today?
+default persistent._mas_should_rain_today = None
+
+#Loading at init 0 because of season functions
+init python in mas_weather:
+    def shouldRainToday():
+        import random
+        import datetime
```

Comment:
should just move the random/datetime imports to the module level at init -20, since you use it in multiple spots.